Modules/posixmodule.c

```diff
@@ -10060,6 +10065,249 @@ os_times_impl(PyObject *module)
+
+        time.CLOCK_REALTIME
+            A settable system-wide real-time clock.
+            If system clock is changed, timer setting need to be updated.
+            See os.TFD_TIMER_ABSTIME and os.TFD_TIMER_CANCEL_ON_SET.
+
+        time.CLOCK_MONOTONIC
+           A nonsettable monotonically increasing clock.
+           Even if system clock is changed, timer setting will be not affected.
+
+        time.CLOCK_BOOTTIME
+           Same as time.CLOCK_MONOTONIC except that it inclues any time that
+           the system is suspended.
+    /
+    *
+    flags: int = 0
+        0 or a bit mask of os.TFD_NONBLOCK or os.TFD_CLOEXEC.
+
+        os.TFD_NONBLOCK
+            If *TFD_NONBLOCK* is set as a flag, read doesn't blocks.
+            If *TFD_NONBLOCK* is not set as a flag, read block until the timer fires.
+
+        os.TFD_CLOEXEC
+            If *TFD_CLOEXEC* is set as a flag, enable the close-on-exec flag
+
+Create and return a timer file descriptor.
+[clinic start generated code]*/
+
+static PyObject *
+os_timerfd_create_impl(PyObject *module, int clockid, int flags)
+/*[clinic end generated code: output=1caae80fb168004a input=f23bfa49fc4b6576]*/
+
+{
+    int fd;
+    Py_BEGIN_ALLOW_THREADS
+    flags |= TFD_CLOEXEC;  // PEP 446: always create non-inheritable FD
+    fd = timerfd_create(clockid, flags);
+    Py_END_ALLOW_THREADS
+    if (fd == -1) {
+        return PyErr_SetFromErrno(PyExc_OSError);
+    }
+    return PyLong_FromLong(fd);
+}
+
+/*[clinic input]
+os.timerfd_settime
+
+    fd: fildes
+        A timer file descriptor.
+    /
+    *
+    flags: int = 0
+        0 or a bit mask of TFD_TIMER_ABSTIME or TFD_TIMER_CANCEL_ON_SET.
+    initial: double = 0.0
+        The initial expiration time, in seconds.
+
+        * If *flags* has TFD_TIMER_ABSTIME bit, *flags* must be in absolute time.
+        * If *flags* doesn't have TFD_TIMER_ABSTIME bit, *flags* must be in relative time.
+        * If *flags* has TFD_TIMER_ABSTIME bit and TFD_TIMER_CANCEL_ON_SET bit and *flags* and time.CLOCK_REALTIME
+          and system clock is changed discontinuously, reading a file descriptor is aborted with ECANCELED.
+    interval: double = 0.0
+        The timer's interval, in seconds.
+
+        * If 'interval' is zero, timer will be run once.
+        * If 'interval' is non-zero, timer will be run periodically with the interval.
+
+Alter a timer file descriptor's internal timer in seconds.
+[clinic start generated code]*/
+
+static PyObject *
+os_timerfd_settime_impl(PyObject *module, int fd, int flags, double initial,
+                        double interval)
+/*[clinic end generated code: output=0dda31115317adb9 input=436a433c2c9664f0]*/
+{
+    struct itimerspec new_value;
+    struct itimerspec old_value;
+    int result;
+    _PyTime_AsTimespec(_PyTime_FromSecondsDouble(initial), &new_value.it_value);
+    _PyTime_AsTimespec(_PyTime_FromSecondsDouble(interval), &new_value.it_interval);
```

Comment:
You must not ignore silently overflow errors: you can write "if (as_timespec() < 0) return NULL;``.

Same in os_timerfd_settime_ns_impl().

Reply:
@vstinner 
Fixeds.